From 18611335c830d6ea78d79e2376559c8f221db1dc Mon Sep 17 00:00:00 2001
From: Ira Cooper <samba@ira.wakeful.net>
Date: Tue, 31 Jan 2012 12:15:36 -0800
Subject: [PATCH] Fix bug #8729 - getpass regressions on Solaris/Illumos - 3.6
 and master.

Autobuild-User: Jeremy Allison <jra@samba.org>
Autobuild-Date: Tue Jan 31 23:28:09 CET 2012 on sn-devel-104
(cherry picked from commit a1901b55cfa658f39a33d0ea120641f56453fe4e)
---
 lib/replace/replace.h       |   11 +++++++++++
 lib/replace/system/passwd.h |    2 ++
 2 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/lib/replace/replace.h b/lib/replace/replace.h
index c081f23..c47cf1c 100644
--- a/lib/replace/replace.h
+++ b/lib/replace/replace.h
@@ -802,4 +802,15 @@ int fdatasync(int );
 /* prototype is in "system/network.h" */
 #endif
 
+#if !defined(getpass)
+#ifdef REPLACE_GETPASS
+#if defined(REPLACE_GETPASS_BY_GETPASSPHRASE)
+#define getpass(prompt) getpassphrase(prompt)
+#else
+#define getpass(prompt) rep_getpass(prompt)
+char *rep_getpass(const char *prompt);
+#endif
+#endif
+#endif
+
 #endif /* _LIBREPLACE_REPLACE_H */
diff --git a/lib/replace/system/passwd.h b/lib/replace/system/passwd.h
index aaea9c8..03476b9 100644
--- a/lib/replace/system/passwd.h
+++ b/lib/replace/system/passwd.h
@@ -67,6 +67,7 @@
 #include <compat.h>
 #endif
 
+#if !defined(getpass)
 #ifdef REPLACE_GETPASS
 #if defined(REPLACE_GETPASS_BY_GETPASSPHRASE)
 #define getpass(prompt) getpassphrase(prompt)
@@ -75,6 +76,7 @@
 char *rep_getpass(const char *prompt);
 #endif
 #endif
+#endif 
 
 #ifndef NGROUPS_MAX
 #define NGROUPS_MAX 32 /* Guess... */
-- 
1.7.7.3

